clicommand/agent_start.go

```diff
@@ -720,6 +721,9 @@ var AgentStartCommand = cli.Command{
 		// Setup the agent pool that spawns agent workers
 		pool := agent.NewAgentPool(workers)
 
+		// Agent-wide shutdown hook. Once per agent, for all workers on the agent.
+		defer func() { shutdownHook(l, cfg.HooksPath) }()
```

Comment:
Could we drop the `func` wrapper here?
I don't think the logger nor the HooksPath could change between `defer` and the call.

```suggestion
		defer shutdownHook(l, cfg.HooksPath)
```